src/index.js

```diff
@@ -39,30 +39,56 @@ loader.pitch = function (request) {
 
   const remainingRequest = JSON.stringify(workerLoader + '!' + request);
 
+  const wrapperBuilder = `
+  var Worker = require(${remainingRequest});
+  var wrap = require('comlink').wrap;
+
+  function getWrappedWorker() {
+    var worker = Worker();
+    var wrapper = wrap(worker);
+
+    if (typeof Proxy === 'function') {
```

Comment:
Could this just use Object.defineProperty to handle both cases? (not sure if the comlink proxy traps it)

Reply:
> Just a couple questions, otherwise this looks good and the tests are a welcome addition.
> 
> I'm mainly concerned with making sure the Proxy usage doesn't reduce browser support - I'm guessing since Comlink already relies on Proxy it's okay?

I'm not that familiar with proxies (I actually discovered them when looking at comlink's code) but from what I understood it traps all attempts to access properties and thus to get a result it has to be defined as a property in the proxied object before creating the proxy or have a custom 'resolver' for a custom property key (as I did in comlink with the 'terminate' method). 

In our case, the wrap function creates and returns a Proxy object over our worker and thus can't provide a '.worker' property accessor as that would mean either the wrapped object (the worker) has a 'worker' property or the proxy object is created with a resolver for 'worker' which would return the proxied object and would, in this case, break the sandboxing contract offered by the Proxy.

So the only solution I found at the time was to proxy the proxy so I could add my custom resolver. 


I just tested with: 

`return Object.defineProperty(wrapper, 'worker', {value: worker});` 

and a dirty hack: 

```
    var worker = Worker();
    worker = Object.defineProperty(worker, 'worker', {value: worker});
    
    var wrapper = wrap(worker);
    return wrapper;
```


The unit tests are failing in both cases (glad I wrote them :-)) with :

TypeError: 'get' on proxy: property 'worker' is a read-only and non-configurable data property on the proxy target but the proxy did not return its actual value (expected '#<Worker>' but got '[object Function]')

Not sure what to do with this error and I found a couple of occurrences of the same error on other projects and StackOverflow.

 
As to the browser support, as you said, the first thing comlink does is create a proxy, so I'm not adding any more constraint. According to https://caniuse.com/?search=Proxy only IE11 and the Baidu Browser 7.12 (released in 2017) do support Web Workers but not Proxy, but I guess as official support of IE11 has been dropped and Baidu Browser seems to be now in version 43 and running on Chromium it should be totally fine to assume we're not causing more harm.